include/eve/arch/cpu/wide.hpp

```diff
@@ -735,13 +735,32 @@ namespace eve
       return detail::self_shl(w, s);
     }
 
+    //! @brief Perform the compound left-shift on all the eve::wide lanes with a constant and assign
+    //! the result to current one.
+    template<typename S, S V>
+    friend EVE_FORCEINLINE auto& operator<<=(wide& w, std::integral_constant<S,V> const& s) noexcept
```

Comment:
Why not eve::index, why constant

Reply:
it's more flexible, index is inheriting from integral constant.
Do we want to keep ppl in eve::index ?

Reply:
Yeah, index is the correct type here